packages/js/components/src/experimental-select-control/select-control.tsx

```diff
@@ -139,6 +139,8 @@ function SelectControl< ItemType = DefaultItemType >( {
 		setInputValue( getItemLabel( singleSelectedItem ) );
 	}, [ singleSelectedItem ] );
 
+	// TODO: for some reason the value in getInputProps is not updated when singleSelectedItem changes
```

Comment:
There is one bug here having to do with the input `value` prop not being updated properly when changing the editing attribute within the modal, at first (it does update when you change a term). It seems to occur due to the `getInputProps()` string for the `value` prop not updating as expected when the `singleSelectedItem` changes.

I updated `downshift` hoping that would address it, but no dice. I haven't dived into it any further, but if you have any insight @louwie17 or @joshuatf let me know!

Reply:
I've resolved this with https://github.com/woocommerce/woocommerce/pull/35269/commits/f5d083979310bd4c2ac7c5645bd6f48c89c38002